iati/version.py

```diff
@@ -0,0 +1,191 @@
+
+    @property
+    def iativer_str(self):
+        """string: An IATIver-format string representation of the Version Number.
+
+        Note:
+            The name of this property may change.
+        """
+        return str(self.integer) + '.0' + str(self.decimal)
+
+    @property
+    def semver_str(self):
+        """string: A SemVer-format string representation of the Version Number.
+
+        Note:
+            The name of this property may change.
+        """
+        return '.'.join([str(self.major), str(self.minor), str(self.patch)])
+
+    def __repr__(self):
+        """str: A representation of the Version Number that will allow a copy of this object to be instantiated."""
+        return "iati.Version('" + self.semver_str + "')"
+
+    def __str__(self):
+        """str: A representation of the Version Number as would exist on the Version Codelist.
+
+        Warning:
+            At present this always results in an IATIver string. This may change should SemVer be adopted.
+            The helper methods must be used if a specific format is required.
+        """
+        return self.iativer_str
+
+    def _is_iatidecimal(self, version):
+        """Determine whether a version string is a Decimal and is a permitted value.
+
+        Args:
+            version (string or Decimal): The value to check conformance of.
+
+        Returns:
+            bool: True if the provided string is a permitted in IATIver-format version number. False if not.
+
+        """
+        if not isinstance(version, Decimal):
+            return False
+
+        valid_values = [Decimal('1.0' + str(val)) for val in range(1, 10)]
+
+        return version in valid_values
+
+    def _is_iativer(self, version_string):
+        """Determine whether a version string is in a IATIver format and is a permitted value.
+
+        Args:
+            version_string (string): The string to check conformance of.
+
+        Returns:
+            bool: True if the provided string is a permitted in IATIver-format version number. False if not.
+
+        """
+        # a regex for what makes a valid IATIver Version Number format string
+        iativer_re = re.compile(r'^((1\.0[1-9])|(((1\d+)|([2-9](\d+)?))\.0[1-9](\d+)?))$')
+
+        return iativer_re.match(version_string)
+
+    def _is_semver(self, version_string):
+        """Determine whether a version string is in a SemVer format and is a permitted value.
+
+        Args:
+            version_string (string): The string to check conformance of.
+
+        Returns:
+            bool: True if the provided string is a permitted in SemVer-format version number. False if not.
+
+        """
+        is_semver_format = semantic_version.validate(version_string)
+        try:
+            is_permitted_value = semantic_version.Version(version_string).major != 0
+        except ValueError:
+            return False
+
+        return is_semver_format and is_permitted_value
+
+    def next_major(self):
```

Comment:
Do we have use cases for the creation of `next_minor|major` ?

Reply:
Primary reason they exist: The underlying SemVer library has them.

They are not currently used within pyIATI.

Reply:
https://github.com/IATI/pyIATI/pull/273/commits/784730962fea36213ffa22d0071dda4064eb372b